hail/src/main/scala/is/hail/rvd/RVDPartitioner.scala

```diff
@@ -69,7 +69,7 @@ class RVDPartitioner(
       Some(Interval(rangeBounds.head.left, rangeBounds.last.right))
 
   def satisfiesAllowedOverlap(testAllowedOverlap: Int): Boolean =
-    RVDPartitioner.isValid(sm, kType, rangeBounds, testAllowedOverlap)
+    (testAllowedOverlap >= kType.size) || RVDPartitioner.isValid(sm, kType, rangeBounds, testAllowedOverlap)
```

Comment:
Could we move the guard into `RVDPartitioner.isValid`?

Reply:
yeah, good change.